### PR TITLE
generate public url through asset listeners

### PR DESCRIPTION
**Pull request by davidhoeck**
## Description of the bug 

The public url displayed in the admin backend of pimcore always displays an url, with the hostname of the Pimcore application as a base. 

If you use AWS S3 or AWS Cloudfront for example, the public url should be the specific url of the used system.

## Changes in this pull request  
This PR generates the public url of an asset via the asset events. 
If the url does not include http (no public url) it will prepend the hostname of the Pimcore application.

### CODE REVIEW

**Comment by brusch** [comment]
@davidhoeck Thanks a lot for your PR! 
I'm just thinking if it would make sense to just add an optional argument to `Asset::getFullPath()` to force returning the frontend path, this would avoid duplicating code. WDYT?

**Comment by davidhoeck** [comment]
@brusch good idea!  I also thought of that, should I update the PR ?

**Comment by brusch** [comment]
@davidhoeck yes, but for BC reasons it's probably better to introduce a new method `getFrontendFullPath()` (tagged as `@internal`)  which is then used in `getFullPath()` and in the controller.

**Comment by davidhoeck** [comment]
@brusch done ✅

File: bundles/AdminBundle/Controller/Admin/AssetController.php [hunk 1]
```diff
@@ -25,6 +25,7 @@
 use Pimcore\Event\Admin\ElementAdminStyleEvent;
 use Pimcore\Event\AdminEvents;
 use Pimcore\Event\AssetEvents;
+use Pimcore\Event\FrontendEvents;
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/AssetController.php:28]:
```suggestion
```

**Comment by brusch** [comment]
Nice, thanks a lot! 👍

**Pull request merged**